### PR TITLE
ALTAPPS-1001: Android fix code autocomplete

### PR DESCRIPTION
**Pull request by XanderZhu**
**YouTrack Issues**:
[ALTAPPS-1001](https://vyahhi.myjetbrains.com/youtrack/issue/ALTAPPS-1001)

**Checklist**

_Before Code Review:_

- [x] Fields "Assignees, Labels, Milestone" are filled in the pull request;
- [x] All checks have been passed;
- [x] Changes have been checked locally.

**Description**
- Add missing languages, like `Java11` and `Java17`
- Move ProgrammingLanguage to the shared module to use in iOS later

### CODE REVIEW

**Review by ivan-magda — APPROVED** [review]

File: shared/src/commonMain/kotlin/org/hyperskill/app/code/domain/model/ProgrammingLanguage.kt [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    GO("go"),
+    PASCAL("pascalabc"),
+    PERL("perl"),
+    SQL("sql"),
+    SWIFT("swift"),
+    PHP("php"),
+    JULIA("julia"),
+    DART("dart");
+
+    val extension: String
```
[the remark this answers is not in the document]
Comment (XanderZhu) [reply shared/src/commonMain/kotlin/org/hyperskill/app/code/domain/model/ProgrammingLanguage.kt:43]: Yes, filename extension.
I'll rename this property.

File: shared/src/commonMain/kotlin/org/hyperskill/app/code/domain/model/ProgrammingLanguage.kt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    companion object {
+        fun of(languageName: String): ProgrammingLanguage? =
+            ProgrammingLanguage.values()
+                .find {
+                    it.languageName.equals(languageName, ignoreCase = true)
+                }
+    }
+}
+
+fun extensionForLanguage(lang: String): String =
```
[the remark this answers is not in the document]
Comment (XanderZhu) [reply shared/src/commonMain/kotlin/org/hyperskill/app/code/domain/model/ProgrammingLanguage.kt:82]: @ivan-magda 
> What purpose usage of this extensionForLanguage on Android?

It is used on Android as a shortcut, to get file extension. A file extension is required for code highlighter.

[the remark this answers is not in the document]
Comment (XanderZhu) [reply shared/src/commonMain/kotlin/org/hyperskill/app/code/domain/model/ProgrammingLanguage.kt:82]: > Maybe it's better to return `String?` here?

I left the old return type, to not change the dependent code. However, it's not a problem to update it.

[the remark this answers is not in the document]
Comment (ivan-magda) [reply shared/src/commonMain/kotlin/org/hyperskill/app/code/domain/model/ProgrammingLanguage.kt:82]: > A file extension is required for code highlighter.

@XanderZhu OK, thank you so much for explaining, changes are not needed.

On iOS we use different code highlighting so I think that these shortcuts may be relevant only for Android, but we will know this while migrating iOS to the `ProgrammingLanguage` from the shared module.

**Pull request merged**